parkhangs-webapp/src/features/comments/commentSlice.js

```diff
@@ -0,0 +1,240 @@
+import { createSlice } from '@reduxjs/toolkit'
+import CommentService from 'services/comment.service'
+
+const commentSlice = createSlice({
+    name: 'comments',
+    initialState: {
+        commentsByParkId: {},
+        commentsByEventId: {},
+        flattenedParkComments: [],
+        flatennedEventsComments: [],
+        loadingParkComments: true,
+        loadingEventComments: false,
+        deletingParkComment: false,
+        deletingEventComment: false,
+        addingComment: false,
+        error : null
+    },
+    reducers: {
+        fetchParkCommentsStart (state) {
+            state.loadingParkComments = true
+            state.error = null
+        },
+
+        fetchParkCommentsSuccessful (state, action) {
+          // if (Object.keys(state.commentsByParkId).length !== 0) {
+          //   state.commentsByParkId = {};
+          // }
+            const events = action.payload
+            state.flattenedParkComments = action.payload
+            const parsedComments = events.reduce(function (acc, comment) {
+                const parkId = comment.parkId.toString()
+                const hasSeenParkSoFar = !!acc[parkId]
+                if (hasSeenParkSoFar) {
+                    acc[parkId].push(comment)
+                }
+                else {
+                    acc[parkId] = [comment]
+                }
+                return acc
+            }, {})
+            state.commentsByParkId = parsedComments
+        },
+
+        fetchParkCommentsFailure (state, action) {
```

Comment:
got halfway throuogh reading this file and realized it's a duplicate of the same file you made at the app level folder 😆  please delete that one and keep this one in the comments feature folder. Please make the changes that I suggested in the app level one!